packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts

```diff
@@ -280,7 +280,10 @@ export async function connectCoinbaseWalletSDK(
   })) as string | number;
 
   const chainId = normalizeChainId(connectedChainId);
-  let chain = defineChain(chainId);
+  let chain =
+    options.chain && options.chain.id === chainId
+      ? options.chain
+      : defineChain(chainId);
   // Switch to chain if provided
   if (
     connectedChainId &&
```

Comment:
Should this be `chainId` and `chainId !== options?.chain.id` so it uses the normalized id?

Reply:
The idea is to use the `options.chain` if provider is connected to `options.chain.id` - I'm not sure what you are proposing here - can you explain?

Reply:
Just to use `chainId` instead of `connectedChainId` on this if statement, sorry it's not part of the actual changes just something I noticed